src/data_structures.rs

```diff
@@ -59,68 +88,59 @@ pub trait PCVerifierKey:
     fn max_degree(&self) -> usize;
 
     /// Outputs the maximum degree supported by the verifier key.
-    fn supported_degree(&self) -> usize;
+    fn segment_size(&self) -> usize;
 
     /// Returns the hash of `self` instance.
     fn get_hash(&self) -> &[u8];
 
     /// Randomize key for testing purpose
-    fn randomize(&mut self);
+    // #[cfg(test)]
+    fn ut_randomize(&mut self);
 }
 
-/// Defines the minimal interface of prepared verifier keys for any polynomial
-/// commitment scheme.
-pub trait PCPreparedVerifierKey<Unprepared: PCVerifierKey> {
-    /// prepare
-    fn prepare(vk: &Unprepared) -> Self;
+/// Defines the minimal interface of the verifier state from succinct
+/// verify part.
+/// This state is returned by the succinct verifier and contains everything
+/// that is needed to complete verification.
+pub trait PCVerifierState: Clone + Debug + Eq + PartialEq {}
+
+/// Defines the minimal interface for opening proofs for a polynomial commitment
+pub trait PCProof:
+    Clone + Debug + Eq + PartialEq + CanonicalSerialize + CanonicalDeserialize + SemanticallyValid
+{
+    /// Return length of committer key
```

Comment:
Not sure how to avoid this. As I remember there was a statement that in common case VerifierKey and ProverKey can have  different sizes. In domain extended succinct verifier we need to know the length of prover key, because it is used for building LC, but have only verifier key here. So the only place from where we can restore the prover key length is the proof.

Domain Extended PC is generic and not a part of implementation, so this function cannot be moved to struct - we need to have it in trait.

So the only way I see is to expand comment, what we should write down here?

Reply:
I don't want to move this function, I would like to remove it. Whenever you need, in the code, to read the size of the key from the proof you'll do it directly there, without the need to call a function to be placed somewhere

Reply:
This function doesn't make sense even in a generic perspective. I removed it.
(See the comment in DomainExtendedPolynomialCommitment::succinct_verify to see how we are going to react on this)